### DISCUSSION TITLE
Searching for new episodes

### DISCUSSION THREAD

**Discussion by joshali1990** [itsToggle/plex_debrid#72] [General]
Is it just me or has the script stopped looking for new releases since the latest update? For example, the script doesn't seem to be searching for the latest episode of better call saul despite it being in my plex watchlist, trakt watchlist and trakt collection. Same with the show the anarchists, which had the latest episode released on 8/14. I checked the ignore list and if the episodes were marked as played in my plex watchlist, but they are not.  I tried re-authorizing trakt, but that didn't seem to help.

**Comment by itsToggle** [comment]
which library service have you selected with the newest update? If you chose trakt as your library service and the episode is marked as "collected" on trakt.tv, the script will assume you have already downloaded it.

In any case, Ill try to reproduce this issue

**Reply by joshali1990** [reply]
Still using plex as my library service. Just downgraded the script and it searched for the newest episodes on my trakt watchlist.

**Reply by itsToggle** [reply]
weird, ill check it out! Thanks!

**Reply by itsToggle** [reply]
alright, found the problem. I removed some lines of code in the latest version, which I thought were obsolete.. turns out they are definetly not :D

**Reply by itsToggle** [reply]
latest comment fixes this issue, very nice catch. It would have taken me a while to even notice! thank you

**Reply by joshali1990** [reply]
That worked, thanks! It's not taking into account the played status on the Plex watchlist now though. I have a show with all past episodes marked as played in the discover more page, but it's still scraping.

**Reply by itsToggle** [reply]
Ive opened an issue regarding this bug and will fix it as soon as possible. It's related to the same update, which elevated the trakt functionality from merely matching trakt media to the plex media type to trakt having its own media type.

**Reply by itsToggle** [reply]
latest commit fixes this issue. Trakt content will now respect the plex ignore feature, if the library service is set to "plex library". If the library service is set to trakt, there is currently no method to ignore content.

**Reply by joshali1990** [reply]
That worked part of the time, but I'm getting a new error:

Exception in thread Thread-913:
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.9/3.9.13_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/threading.py", line 980, in _bootstrap_inner
    self.run()
  File "/opt/homebrew/Cellar/python@3.9/3.9.13_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/threading.py", line 917, in run
    self._target(*self._args, **self._kwargs)
  File "/Volumes/external/plex_rd.py", line 2938, in download
    result[index] = cls.download(library=library,parentReleases=parentReleases)
  File "/Volumes/external/plex_rd.py", line 414, in download
    episode.watch()
  File "/Volumes/external/plex_rd.py", line 163, in watch
    url = 'https://metadata.provider.plex.tv/actions/scrobble?identifier=tv.plex.provider.metadata&key='+self.ratingKey+'&X-Plex-Token='+ plex.users[0][1]
AttributeError: 'episode' object has no attribute 'ratingKey'

**Reply by itsToggle** [reply]
okay, ive just released a fix for this. Trakt content can now be ignored on plex by plex_debrid and the "ignored media" option in the main menu should now work as expected for trakt content aswell.

Sorry for the mess, the program is now a whole lot bigger than what Im used to and catching every error is pretty hard to do on my own. Thanks for your testing!

**Reply by joshali1990** [reply]
That worked thanks! Thanks again for your prompt response.

**Comment by ineedscissors61** [comment]
I'm not using Trakt and I'm not getting any newly released episodes. any suggestions?

**Reply by itsToggle** [reply]
Are you on the latest version of this script? I've released a fix for this bug a few hours ago

**Reply by ineedscissors61** [reply]
yeah i'm on the latest version

**Reply by itsToggle** [reply]
Are the show/s in question inside your plex watchlist? Have you made sure that the new episode is not marked as "watched" inside Plex Discover? plex_debrid uses the Plex Discover watched status as a way to ignore content that couldn't be found.

**Reply by ineedscissors61** [reply]
ive double checked this and they are not marked as "played"
I ave had a look at the "ignored media" in plex debrid and the shows are not in there either

**Reply by ineedscissors61** [reply]
ive just removed one of the shows from my watch list and added it again and its started downloading

**Reply by itsToggle** [reply]
That's an odd behaviour. plex_debrid completely refreshes the entire watchlist every 30 minutes, so adding something back to the watchlist should not change anything. I'll go through the code again.

**Reply by ineedscissors61** [reply]
I thought i should let you know that shows aren't downloading again. it seems removing and adding back to my watchlist just forced the episodes to download rather than fix the problem
